controllers/managedocs_controller.go

```diff
@@ -210,21 +217,29 @@ func (r *ManagedOCSReconciler) setStorageClusterParamsFromSecret(sc *ocsv1.Stora
 	if err != nil {
 		return fmt.Errorf("Invalid storage cluster size value: %v", sizeAsString)
 	}
-
 	r.Log.Info("Storage cluster parameters", "count", sdsCount)
 
 	var ds *ocsv1.StorageDeviceSet = nil
-	for _, item := range sc.Spec.StorageDeviceSets {
+	for index, item := range sc.Spec.StorageDeviceSets {
 		if item.Name == deviceSetName {
-			ds = &item
+			ds = &sc.Spec.StorageDeviceSets[index]
 			break
 		}
 	}
```

Comment:
Using an item will create an in-memory clone of each item in the array, which in this case is a `DeviceSet` object. 
Because you already need the index I will suggest the following to prevent in-memory cloning:

```
	for index := range sc.Spec.StorageDeviceSets {
                 item := &sc.Spec.StorageDeviceSets[index]
		if item.Name == deviceSetName {
			ds = item
			break
		}
	}
```